include/aws/iotdevice/private/secure_tunneling_impl.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#ifndef AWS_IOTDEVICE_SECURE_TUNNELING_IMPL_H
+#define AWS_IOTDEVICE_SECURE_TUNNELING_IMPL_H
+
+#include <aws/iotdevice/secure_tunneling.h>
+
+#include <aws/common/condition_variable.h>
+#include <aws/common/mutex.h>
+#include <aws/io/tls_channel_handler.h>
+
+struct data_tunnel_pair {
+    struct aws_byte_buf buf;
+    struct aws_byte_cursor cur;
+    const struct aws_secure_tunnel *secure_tunnel;
+    bool length_prefix_written;
+};
+
+struct aws_secure_tunnel_vtable {
+    int (*connect)(struct aws_secure_tunnel *secure_tunnel);
+    int (*send_data)(struct aws_secure_tunnel *secure_tunnel, const struct aws_byte_cursor *data);
+    int (*send_stream_start)(struct aws_secure_tunnel *secure_tunnel);
+    int (*send_stream_reset)(struct aws_secure_tunnel *secure_tunnel);
+    int (*close)(struct aws_secure_tunnel *secure_tunnel);
+};
+
+struct aws_secure_tunnel {
+    /* Static settings */
+    struct aws_allocator *alloc;
+    struct aws_secure_tunnel_options_storage *options_storage;
+    struct aws_secure_tunnel_options *options;
```

Comment:
Why do we still need this pointer if the struct is private and we have an options embedded in the storage?